manim/animation/override_animation.py

```diff
@@ -0,0 +1,102 @@
+"""Decorator for Mobject-specific Animation overrides."""
+
+
+__all__ = ["override_animation", "_setup_animation_overriding"]
+
+from typing import Callable, Dict, List, Type
+
+from manim.mobject.mobject import Mobject
+from manim.utils.exceptions import MultiAnimationOverrideException
+
+from .. import logger
+
+
+def _setup_animation_overriding(
+    mobject_class: "Type(Mobject)",
+    overrides: "Dict[Type[Animation], List[Dict[Type[Mobject], str]]]",
+):
+    """Sets up :class:`~.Animation` overrides for a :class:`~.Mobject` subclass.
+
+    Must be called to enable the functionality of methods decorated with ``@``
+    :func:`override_animation`.
+
+    Parameters
+    ----------
+    mobject_class
+        The Mobject subclass to add an Animation override to.
+    overrides
+        The overrides dict to add to.
+
+    Raises
+    ------
+    MultiAnimationOverrideException
+        If one class defines multiple overrides for a single ``Animation``.
+
+    """
+    for method_name in dir(mobject_class):
+        if method_name.startswith("__"):  # Preventing attribute errors
+            continue
+        method = getattr(mobject_class, method_name)
+        if hasattr(method, "_override_animation"):
+            animation_class = method._override_animation
+
+            if animation_class not in overrides:
+                overrides[animation_class] = {}
+            if mobject_class in overrides[animation_class]:
+                msg = (
+                    f"The animation {animation_class.__name__} for "
+                    f"{mobject_class.__name__} is overridden by more than one method: "
+                    f"{overrides[animation_class][mobject_class].__qualname__}"
+                    f" and {method.__qualname__}. If one of these methods is "
+                    f"inherited make sure they are named equally."
+                )
+                logger.error(msg)
+                raise MultiAnimationOverrideException()
```

Comment:
it's better to simply raise the msg.
```suggestion
                raise MultiAnimationOverrideException(msg)
```